js/modules/k6/experimental/tracing/trace_id.go

```diff
@@ -0,0 +1,76 @@
+package tracing
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+)
+
+const (
+	// Being 075 the ASCII code for 'K' :)
+	k6Prefix = 0o756
+
+	// To ingest and process the related spans in k6 Cloud.
+	k6CloudCode = 12
+
+	// To not ingest and process the related spans, b/c they are part of a non-cloud run.
+	k6LocalCode = 33
+)
+
+// TraceID represents a trace-id as defined by the [W3c specification], and
+// used by w3c, b3 and jaeger propagators. See [Considerations for trace-id field generation]
+// for more information.
+//
+// [W3c specification]: https://www.w3.org/TR/trace-context/#trace-id
+// [Considerations for trace-id field generation]: https://www.w3.org/TR/trace-context/#considerations-for-trace-id-field-generation
+//
+//nolint:lll
+type TraceID struct {
+	Prefix            int16
+	Code              int8
+	UnixTimestampNano uint64
+}
+
+// NewTraceID returns a new TraceID with the given prefix, code and unix timestamp in nanoseconds.
+func NewTraceID(prefix int16, code int8, unixTimestampNano uint64) TraceID {
+	return TraceID{
+		Prefix:            prefix,
+		Code:              code,
+		UnixTimestampNano: unixTimestampNano,
+	}
+}
+
+// Encode encodes the TraceID into a hex string and a byte slice.
+func (t TraceID) Encode() (string, []byte, error) {
+	if !t.isValid() {
+		return "", nil, fmt.Errorf("failed to encode traceID: %v", t)
+	}
+
+	buf := make([]byte, 16)
+
+	n := binary.PutVarint(buf, int64(t.Prefix))
```

Comment:
You could use https://pkg.go.dev/encoding/binary#ByteOrder

Reply:
I must admit I'm not too familiar with the binary package, so I don't necessarily see neither how to do the same with the `ByteOrder` interface nor how it would benefit the implementation over `PutVarint`/`PutUvarint`. Would you be able to illustrate it with an example? 😃

Reply:
There is an example in the linked page. Let me know if it is not enough.

Reply:
Hehe, I've seen 👀 I even experimented with it, but it was somewhat inconclusive, and I had a hard time reproducing the current behavior. Hence me asking for your input as it sounded like you might have something specific in mind. 👍🏻

Reply:
Re-checking better this, I've just realized that Prefix is signed int16. Does it have a specific reason?

Anyway, this is not a blocker, it is mostly a minor optimization for the feature.